lib/workers/repository/update/branch/reuse.ts

```diff
@@ -122,16 +122,22 @@ async function determineRebaseWhenValue(
   result: BranchConfig,
   keepUpdated: boolean,
 ): Promise<void> {
-  if (result.rebaseWhen === 'auto') {
+  if (result.rebaseWhen === 'auto' || result.rebaseWhen === 'automerging') {
     let reason;
 
     let newValue = 'behind-base-branch';
     if (result.automerge === true) {
       reason = 'automerge=true';
-    } else if (await platform.getBranchForceRebase?.(result.baseBranch)) {
+    } else if (
+      result.rebaseWhen !== 'automerging' &&
+      (await platform.getBranchForceRebase?.(result.baseBranch))
+    ) {
       reason = 'platform is configured to require up-to-date branches';
     } else if (keepUpdated) {
       reason = 'keep-updated label is set';
+    } else if (result.rebaseWhen === 'automerging') {
+      newValue = 'never';
+      reason = 'no keep-updated label and automerging is set';
```

Comment:
Move this clause earlier so that you don't need `result.rebaseWhen !== 'automerging' &&` above

Reply:
done